packages/suite/src/components/suite/ToastNotification/index.tsx

```diff
@@ -30,73 +15,96 @@ const Wrapper = styled.div<Pick<ViewProps, 'variant'>>`
 `;
 
-const StyledButton = styled(Button)`
-    margin-left: 10px;
+const StyledButton = styled(Button)<Pick<ViewProps, 'action'>>`
+    ${props =>
+        (!props.action?.position || props.action.position === 'right') &&
+        css`
+            margin-left: 10px;
+        `};
+    ${props =>
+        props.action?.position === 'bottom' &&
+        css`
+            margin-top: 8px;
+            font-weight: ${variables.FONT_SIZE.NORMAL};
+            height: 32px;
+        `};
 `;
 
-const CancelWrapper = styled.div`
+const StyledCancelIcon = styled(Icon)`
     margin-left: 10px;
-    margin-top: 6px;
-    align-self: flex-start;
 `;
 
 const ToastNotification = ({
     icon,
     message,
     action,
-    actionLabel,
+    variant,
     cancelable = true,
+    onCancel,
     ...props
 }: ViewProps) => {
-    const defaultIcon = icon ?? getNotificationIcon(props.variant);
+    const defaultIcon = icon ?? getNotificationIcon(variant);
+
     const { closeNotification } = useActions({
         closeNotification: notificationActions.close,
     });
 
     const dataTestBase = `@toast/${props.notification.type}`;
 
+    const actionButton = action && (
+        <StyledButton
+            variant={action.variant || 'tertiary'}
+            onClick={action.onClick}
+            action={action}
+            fullWidth
+        >
+            <Translation id={action.label} />
+        </StyledButton>
+    );
+
     return (
-        <Wrapper data-test={dataTestBase} variant={props.variant}>
-            {defaultIcon && (
-                <Icon icon={defaultIcon} size={20} color={getVariantColor(props.variant)} />
-            )}
-            <Title>
-                {typeof message === 'string' ? (
-                    <Translation id={message} />
-                ) : (
-                    <Translation {...message} />
-                )}
-            </Title>
-            {action && actionLabel && (
-                <StyledButton
-                    variant="tertiary"
-                    icon="ARROW_RIGHT"
-                    alignIcon="right"
-                    onClick={action}
-                >
-                    <Translation id={actionLabel} />
-                </StyledButton>
+        <Wrapper data-test={dataTestBase} variant={variant}>
+            {defaultIcon && typeof defaultIcon === 'string' ? (
+                <Icon icon={defaultIcon} size={20} color={getVariantColor(variant)} />
+            ) : (
+                defaultIcon
             )}
+            <BodyWrapper>
+                <Message>
+                    {typeof message === 'string' ? (
+                        <Translation id={message} />
+                    ) : (
+                        <Translation {...message} />
+                    )}
+                </Message>
+                {(action?.position === 'bottom' || !action?.position) && actionButton}
+            </BodyWrapper>
+            {(action?.position === 'right' || !action?.position) && actionButton}
```

Comment:
what position is the default one? 'bottom' I guess? But if `!action?.position` is true (position is undefined), actionButton is rendered twice, right?

```suggestion
            {action?.position === 'right' && actionButton}
```

Reply:
The default one is `right`.

Reply:
221bb89aab7e03dfb7e468ccda48d491067d0054